MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -375,6 +375,11 @@
 				portabilityFeatures->vertexAttributeAccessBeyondStride = true;	// Costs additional buffers. Should make configuration switch.
 				break;
 			}
+			case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_INTEGER_DOT_PRODUCT_FEATURES: {
+				auto* shaderIntDotFeatures = (VkPhysicalDeviceShaderIntegerDotProductFeatures*)next;
+				shaderIntDotFeatures->shaderIntegerDotProduct = true;
+				break;
+			}
```

Comment:
Anytime there is a entry in this chain, a corresponding entry is required in `MVKDeviceFeatureStructs.def`, so that the features can be enabled by the app.

We haven't found any effective and convenient way to assert this yet. If you have any ideas, happy to hear them.

For now, there's just a comment to that effect:


    // For consistency and ease of admin, keep the following list in the same order as in MVKDeviceFeatureStructs.def